roles/install-etcd/tasks/main.yml

```diff
@@ -0,0 +1,36 @@
+- name: Download and setup etcd
+  unarchive:
+    src: "https://storage.googleapis.com/etcd/{{ etcd_version }}/etcd-{{ etcd_version }}-linux-ppc64le.tar.gz"
+    dest: "/usr/local/bin"
+    remote_src: yes
+    extra_opts:
+      - --strip-components=1
+  retries: 3
+  delay: 5
+
+- name: Create the etcd.service file on nodes.
+  copy:
+    src: etcd.service
+    dest: /etc/systemd/system/etcd.service
+    mode: '0644'
+
+- name: Enable and start etcd server on node
+  systemd:
+    name: etcd
+    state: started
+    enabled: True
+
+- name: Disable SWAP since kubernetes can't work with swap enabled (1/2)
+  shell: |
+    swapoff -a
+
+- name: Disable SWAP in fstab since kubernetes can't work with swap enabled (2/2)
+  replace:
+    path: /etc/fstab
+    regexp: '^([^#].*?\sswap\s+sw\s+.*)$'
+    replace: '# \1'
+
+- name: Disable SELinux
+  shell: |
+    setenforce 0
+    sed -i 's/^SELINUX=.*/SELINUX=permissive/g' /etc/selinux/config
```

Comment:
these are generic one, nothing to do with etcd, you may consider handling in a separate role.